crates/ra_hir/src/semantics.rs

```diff
@@ -403,6 +416,35 @@ impl<'db> SemanticsImpl<'db> {
         self.analyze(param.syntax()).type_of_self(self.db, &param)
     }
 
+    fn method_receiver_kind(&self, call: &ast::MethodCallExpr) -> Option<SelfKind> {
+        self.resolve_method_call(call).and_then(|func| {
+            let lookup = func.lookup(self.db.upcast());
+            let src = lookup.source(self.db.upcast());
+            let param_list = src.value.param_list()?;
+            let self_param = param_list.self_param()?;
+            if self_param.amp_token().is_some() {
+                return Some(if self_param.mut_token().is_some() {
+                    SelfKind::Mutable
+                } else {
+                    SelfKind::Shared
+                });
+            }
+
+            let ty = self.type_of_self(&self_param)?;
+            let krate = Function::from(func).krate(self.db)?;
+            let lang_item = self.db.lang_item(krate.id, SmolStr::new("copy"));
```

Comment:
Am I using the right crate here in the `lang_item` lookup? It works in the unit test, but I'm not sure if this is valid for real use cases.